File: TeadsSDKDemo/app/src/main/java/tv/teads/teadssdkdemo/utils/BaseFragment.java
package tv.teads.teadssdkdemo.utils;

import android.support.v4.app.Fragment;

import org.greenrobot.eventbus.EventBus;

import tv.teads.teadssdkdemo.MainActivity;

/**
 * The base fragment
 * Created by Hugo Gresse on 03/04/15.
 */
public abstract class BaseFragment extends Fragment {

    /**
     * Return the pid from activity
     *
     * @return the pid
     */
    protected int getPid() {
        MainActivity activity = (MainActivity) getActivity();
        if(activity == null) return 0;
        return activity.getPid(this.getActivity());
    }

    /**
     * Return the webview url to display
     *
     * @return an url
     */
    public String getWebViewUrl() {
        MainActivity activity = (MainActivity) getActivity();
        if(activity == null) return "";
        return activity.getWebViewUrl(this.getActivity());
    }

    @Override
    public void onResume() {
        super.onResume();
        EventBus.getDefault().register(this);
    }

    @Override
    public void onPause() {
        super.onPause();
        EventBus.getDefault().unregister(this);
    }
}

File: TeadsSDKDemo/app/src/main/java/tv/teads/teadssdkdemo/utils/BaseFragment.kt
package tv.teads.teadssdkdemo.utils

import android.os.Bundle
import android.view.Menu
import android.view.MenuInflater
import androidx.fragment.app.Fragment
import tv.teads.teadssdkdemo.MainActivity
import tv.teads.teadssdkdemo.data.FormatType
import tv.teads.teadssdkdemo.data.PidStore
import tv.teads.teadssdkdemo.data.ProviderType


/**
 * The base fragment
 */
abstract class BaseFragment : Fragment() {

    /**
     * Return the pid
     *
     */
    val pid: Int
        get() {
            return PidStore.getPid(requireContext(), selectedFormat)
        }

    /**
     * Return current selected format type
     *
     */
    var selectedFormat: FormatType = FormatType.INREAD

    /**
     * Return current selected provider type
     *
     */
    var selectedProvider: ProviderType = ProviderType.DIRECT

    /**
     * Return the webview url to display
     *
     * @return an url
     */
    val webViewUrl: String
        get() {
            val activity = activity as MainActivity? ?: return ""
            return activity.getWebViewUrl()
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(true)
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        super.onCreateOptionsMenu(menu, inflater)
        menu.clear()
    }

    abstract fun getTitle(): String
}
